pkg/controller/populators/clone-populator.go

```diff
@@ -472,3 +478,18 @@ func getSavedCloneStrategy(obj client.Object) *cdiv1.CDICloneStrategy {
 func setSavedCloneStrategy(obj client.Object, strategy cdiv1.CDICloneStrategy) {
 	cc.AddAnnotation(obj, cc.AnnCloneType, string(strategy))
 }
+
+func shouldSkipCleanup(pvc *corev1.PersistentVolumeClaim) bool {
+	// We can skip cleanup to keep objects for debugging purposes if:
+	//   - AnnPodRetainAfterCompletion annotation is set to true. This means that the user explicitly wants
+	//     to keep the pods.
+	//   - Clone is host-assisted, which is the only clone type with transfer pods worth debugging.
+	//   - Clone occurs in a single namespace, so we avoid retaining pods in a namespace we don't have right to access.
+	dataSourceNamespace, ok := pvc.Annotations[AnnDataSourceNamespace]
```

Comment:
Let's future proof this for when cross namespace datasource goes live.  In that case `pvc.spec.dataCource.Namespace` may be set